bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/CommonInterface.java

```diff
@@ -0,0 +1,236 @@
+            @Nullable String thingStatusReason);
+
+    void triggerChannel(String channelID, String event);
+
+    void updateThing(Thing thing);
+
+    @Nullable
+    Bridge getBridge();
+
+    default @Nullable CommonInterface getBridgeHandler() {
+        Bridge bridge = getBridge();
+        return bridge != null && bridge.getHandler() instanceof DeviceHandler ? (DeviceHandler) bridge.getHandler()
+                : null;
+    }
+
+    default @Nullable ApiBridgeHandler getRootBridge() {
+        Bridge bridge = getBridge();
+        BridgeHandler bridgeHandler = null;
+        if (bridge != null) {
+            bridgeHandler = bridge.getHandler();
+            while (bridgeHandler != null && !(bridgeHandler instanceof ApiBridgeHandler)) {
+                bridge = ((CommonInterface) bridgeHandler).getBridge();
+                bridgeHandler = bridge != null ? bridge.getHandler() : null;
+            }
+        }
+        return (ApiBridgeHandler) bridgeHandler;
+    }
+
+    default Optional<NetatmoServlet> getServlet() {
+        ThingHandler handler = getThing().getHandler();
+        Bridge root = null;
+        if (handler instanceof ModuleHandler) {
+            CommonInterface bridgeHandler = ((ModuleHandler) handler).getBridgeHandler();
+            if (bridgeHandler != null) {
+                root = bridgeHandler.getBridge();
+            }
+        } else if (handler instanceof DeviceHandler) {
+            root = ((DeviceHandler) handler).getBridge();
+        }
+        ThingHandler rootHandler = root != null ? root.getHandler() : null;
+        if (rootHandler instanceof ApiBridgeHandler) {
+            return ((ApiBridgeHandler) rootHandler).getServlet();
+        }
+        return Optional.empty();
+    }
+
+    default @Nullable String getBridgeId() {
+        CommonInterface bridge = getBridgeHandler();
+        return bridge != null ? bridge.getId() : null;
+    }
+
+    default void expireData() {
+        getCapabilities().values().forEach(cap -> cap.expireData());
+    }
+
+    default String getId() {
+        return (String) getThing().getConfiguration().get("id");
+    }
+
+    default Stream<Channel> getActiveChannels() {
+        return getThing().getChannels().stream()
+                .filter(channel -> ChannelKind.STATE.equals(channel.getKind()) && isLinked(channel.getUID()));
+    }
+
+    default Optional<CommonInterface> getHomeHandler() {
+        CommonInterface bridgeHandler = getBridgeHandler();
+        if (bridgeHandler != null) {
+            return bridgeHandler.getCapabilities().get(HomeCapability.class).isPresent() ? Optional.of(bridgeHandler)
+                    : Optional.empty();
+        }
+        return Optional.empty();
+    }
+
+    default List<CommonInterface> getActiveChildren() {
+        Thing thing = getThing();
+        if (thing instanceof Bridge) {
+            return ((Bridge) thing).getThings().stream().map(Thing::getHandler).filter(Objects::nonNull)
```

Comment:
Please also check that the thing is not disabled

Reply:
Done